mm2src/coins/lp_coins.rs

```diff
@@ -708,7 +716,12 @@ pub trait SwapOps {
         input: SearchForSwapTxSpendInput<'_>,
     ) -> Result<Option<FoundSwapTxSpend>, String>;
 
-    async fn extract_secret(&self, secret_hash: &[u8], spend_tx: &[u8]) -> Result<Vec<u8>, String>;
+    async fn extract_secret(
+        &self,
+        secret_hash: &[u8],
+        spend_tx: &[u8],
+        watcher_reward: bool,
+    ) -> Result<Vec<u8>, String>;
```

Comment:
We started migrating from `ERR`(string errors) couple months ago. I think we should stop adding `String` as an error type to functions.

Reply:
I didn't add this, it was this way already. There are many functions like this with different implementations for all coins. We can open a refactoring issue for this and assign it to someone in next sprints since it will take some time.